src/ApiCore/Transport/GrpcFallbackTransport.php

```diff
@@ -0,0 +1,202 @@
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+namespace Google\ApiCore\Transport;
+
+use Google\ApiCore\ApiException;
+use Google\ApiCore\ApiStatus;
+use Google\ApiCore\Call;
+use Google\ApiCore\ServiceAddressTrait;
+use Google\ApiCore\ValidationException;
+use Google\ApiCore\ValidationTrait;
+use Google\Protobuf\Internal\Message;
+use Google\Rpc\Status;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7\Request;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * A transport that sends protobuf over HTTP 1.1 that can be used when full gRPC support
+ * is not available.
+ */
+class GrpcFallbackTransport implements TransportInterface
+{
+    use ValidationTrait;
+    use ServiceAddressTrait;
+    use HttpUnaryTransportTrait;
+
+    private $baseUri;
+
+    /**
+     * @param string $baseUri
+     * @param callable $httpHandler A handler used to deliver PSR-7 requests.
+     */
+    public function __construct(
+        $baseUri,
+        callable $httpHandler
+    ) {
+        $this->baseUri = $baseUri;
+        $this->httpHandler = $httpHandler;
+        $this->transportName = 'grpc-fallback';
+    }
+
+    /**
+     * Builds a GrpcFallbackTransport.
+     *
+     * @param string $serviceAddress
+     *        The address of the API remote host, for example "example.googleapis.com".
+     * @param array $config {
+     *    Config options used to construct the grpc-fallback transport.
+     *
+     *    @type callable $httpHandler A handler used to deliver PSR-7 requests.
+     * }
+     * @return GrpcFallbackTransport
+     * @throws ValidationException
+     */
+    public static function build($serviceAddress, array $config = [])
+    {
+        $config += [
+            'httpHandler'  => null,
+        ];
+        list($baseUri, $port) = self::normalizeServiceAddress($serviceAddress);
```

Comment:
I noticed this is the same behavior as in the `RestTransport`. I'm trying to remember why we split the port out here - it seems like it could cause an issue since we aren't passing it down into the constructed transport. We may just want to update line 92 to `return new GrpcFallbackTransport("$baseUri:$port", $httpHandler);`. WDYT?

Reply:
Hmm, I'm not sure. Right now we don't support an alternate port for those transports - if we want to pass it into the constructor, then we should also add support by plumbing it into guzzle, right? Maybe this could be flagged as a separate issue to be fixed?